blog/_posts/2018-06-10-bsp.md

```diff
@@ -0,0 +1,214 @@
+tools that Bloop currently supports--- can try this prototype out.
+
+[BSP v1.0.0](https://github.com/scalacenter/bsp/blob/master/docs/bsp.md)
+has been designed to support the most basic language integration in an
+IDE/editor. IDEs can find all the modules defined in a workspace, ask for
+their dependencies, compile, test, run, and get notifications on modified
+build targets, among a few other features.
+
+Our IntelliJ-Bloop implementation doesn't yet implement all these endpoints.
+Instead, it focuses on providing a faster integration than the traditional
+project import. You can now import your project, browse Scala code with full
+language support and get compiler diagnostics from the build tool in
+IntelliJ's editor.
+
+How does this work? First, you export your project from your build to Bloop.
+This process takes usually ~10s for a medium-sized project (highly
+recommended to use [coursier] in your build for upmost speed) and exists so
+that Bloop can understand different build tools. Then, the IDE imports it
+from Bloop via BSP. In total, the "Import Project" process takes around ~15
+seconds.
+
+You can see it on this demo importing [akka/akka](https://github.com/akka/akka/):
+
+![Demo of BSP import for akka/akka](/resources/img/bloop-install.gif)
+
+In my experience, importing a build via BSP is substantially faster than
+doing so via the traditional sbt or Maven project import where the process
+can take minutes. It also enables smarter communication between build tools
+and IDEs because the communication is bidirectional.
+
+Further, this two-step process can be improved in the future. First, bloop
+could run `bloopInstall` every time you start up your sbt shell or `reload`
+your build. Second, Bloop's Build Server could send notifications the IDE
+whenever there is a change in the build. These enhancements would improve the
+state-of-the-art in IDE project import by not requiring users to manually
+check when the build and IDE have gone off-sync and ask for a "Reload
+Project". If sbt implemented BSP, an sbt shell would suffice to also support
+a project import with these properties instantly.
+
+Aside from importing your build, the BSP prototype also allows you to get
+compiler diagnostics directly from your build tool instead of diagnostics
+from IntelliJ's presentation compiler. Build tool diagnostics are always
+correct since the compilation happens in your build tool and the results
+are streamed to the IDE.
+
+![Demo of BSP compile for akka/akka](/resources/img/bloop-compile.gif)
+
+The BSP integration is now available under an experimental mode in the latest
+IntelliJ 2018.2 nightly. To try the prototype out, follow the instructions in
+[this
+document](https://gist.github.com/jvican/0d738cc0aa47afe5dbbdfe6ea1bd8694).
+
+To learn more about the Build Server Protocol, check out the
+[slides](http://jorge.vican.me/slides/BSP.pdf) of my talk with Justin at
+Scalasphere 2018.
```

Comment:
I still think a lot of high-level information is missing. What's the relationship between this bsp artifact that you link to and bloop? These are two artifacts and if I read this pretending I've never heard of this work before, it's not clear to me how these things are related and why you're talking about this weird thing called Bloop 😉